### PR TITLE
Deprecate old ways of getting AUTOPILOT_VERSION

### PR DESCRIPTION
**Pull request by peterbarker**
We've been able to get `AUTOPILOT_VERSION` using `REQUEST_MESSAGE` for 3-ish years now.

Removing both of these will save ~200 bytes

PR for MAVProxy is here: https://github.com/ArduPilot/MAVProxy/pull/1115
PR for MissionPlanner is here: https://github.com/ArduPilot/MissionPlanner/pull/2989
PR for APM Planner2 is here: https://github.com/ArduPilot/apm_planner/pull/1268
My reading of the QGC source code is that they are already using `MAV_CMD_REQUEST_MESSAGE` for `AUTOPILOT_VERSION`

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -3725,9 +3725,12 @@ void GCS_MAVLINK::handle_common_message(const mavlink_message_t &msg)
         handle_set_mode(msg);
         break;
 
+#if AP_MAVLINK_AUTOPILOT_VERSION_REQUEST_ENABLED
     case MAVLINK_MSG_ID_AUTOPILOT_VERSION_REQUEST:
+        send_received_message_deprecation_warning("AUTOPILOT_VERSION_REQUEST");
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:3730]: this needs to be changed from WARNING to INFO

**Review by tridge — CHANGES_REQUESTED** [review]
we should consider showing these deprecation warnings only on latest and beta, not stable

**Comment by tridge** [comment]
needs testing on a qgc tablet

**Comment by tridge** [comment]
@bugobliterator please test with herelink

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: This is already done in master

**Comment by peterbarker** [comment]
On reflection I've just put the defines in and not the deprecation warnings.

We'll come back and do the warnings later...

**Review by tridge — APPROVED** [review]

**Pull request merged**